packages/cdktf/lib/testing/__tests__/matchers.test.ts

```diff
@@ -28,22 +29,93 @@ function corruptSynthesizedStack(stackPath: string) {
   );
 }
 
+describe("deepEqualIgnoringObjectCasing", () => {
+  it("should compare simple items", () => {
+    expect(deepEqualIgnoringObjectCasing(1, 1)).toBe(true);
+    expect(deepEqualIgnoringObjectCasing("1", "1")).toBe(true);
+    expect(deepEqualIgnoringObjectCasing(true, true)).toBe(true);
+    expect(deepEqualIgnoringObjectCasing(false, false)).toBe(true);
+    expect(deepEqualIgnoringObjectCasing(null, null)).toBe(true);
+    expect(deepEqualIgnoringObjectCasing(undefined, undefined)).toBe(true);
+    expect(deepEqualIgnoringObjectCasing([1, 2, 3], [1, 2, 3])).toBe(true);
+    expect(deepEqualIgnoringObjectCasing({ a: 1, b: 2 }, { a: 1, b: 2 })).toBe(
+      true
+    );
+
+    expect(deepEqualIgnoringObjectCasing(1, 2)).toBe(false);
+  });
+
+  it("should compare arrays", () => {
+    expect(deepEqualIgnoringObjectCasing([1, 2, 3], [1, 2, 3])).toBe(true);
+
+    expect(deepEqualIgnoringObjectCasing([1, 2, 3], [1, 2, 4])).toBe(false);
+    expect(deepEqualIgnoringObjectCasing([1, 2, 3], [1, 2, 3, 4])).toBe(false);
+  });
+
+  it("should compare objects", () => {
+    expect(deepEqualIgnoringObjectCasing({ a: 1, b: 2 }, { a: 1, b: 2 })).toBe(
+      true
+    );
+    expect(
+      deepEqualIgnoringObjectCasing(
+        { a: { c: 3 }, b: 2 },
+        { a: { c: 3 }, b: 2 }
+      )
+    ).toBe(true);
+
+    expect(
+      deepEqualIgnoringObjectCasing({ a: 3, b: 2 }, { a: { c: 3 }, b: 2 })
+    ).toBe(false);
+    expect(
+      deepEqualIgnoringObjectCasing({ a: { c: 3 }, b: 2 }, { a: 3, b: 2 })
+    ).toBe(false);
+  });
+
+  it("should ignore case when comparing object keys", () => {
+    expect(
+      deepEqualIgnoringObjectCasing({ a: 1, fooBar: 2 }, { a: 1, foo_bar: 2 })
+    ).toBe(true);
+    expect(
+      deepEqualIgnoringObjectCasing({ a: { fooBar: 2 } }, { a: { foo_bar: 2 } })
+    ).toBe(true);
+  });
+
+  it("should ignore keys not present in expectation", () => {
+    expect(
+      deepEqualIgnoringObjectCasing(
+        { a: { fooBar: 2 } },
+        { a: { foo_bar: 2 }, b: 3 }
+      )
+    ).toBe(true);
```

Comment:
I would not have expected this for something called `deepEqual` (even if ignoring casing).
I don't need this to be changed, just wanted to note 😄